cmd/osbuild-dnf-json-tests/main_test.go

```diff
@@ -54,3 +59,42 @@ func TestFetchChecksum(t *testing.T) {
 	assert.Nilf(t, err, "Failed to fetch checksum: %v", err)
 	assert.NotEqual(t, "", c["repo"], "The checksum is empty")
 }
+
+// This test loads all the repositories available in /repositories directory
+// and tries to run depsolve for each architecture. With N architectures available
+// this should run cross-arch dependency solving N-1 times.
+func TestCrossArchDepsolve(t *testing.T) {
+	// Set up temporary directory for rpm/dnf cache
+	dir, err := ioutil.TempDir("/tmp", "rpmmd-test-")
+	require.Nilf(t, err, "Failed to create tmp dir for depsolve test: %v", err)
+	defer os.RemoveAll(dir)
+	rpm := rpmmd.NewRPMMD(dir)
+
+	// Load repositories from the definition we provide in the RPM package
+	repositories := "/usr/share/osbuild-composer"
+
+	// TODO: add RHEL, but don't make it hard requirement because it will fail outside of VPN
+	for _, distroStruct := range []distro.Distro{fedora30.New(), fedora31.New(), fedora32.New()} {
+		repoConfig, err := rpmmd.LoadRepositories([]string{repositories}, distroStruct.Name())
+		assert.Nilf(t, err, "Failed to LoadRepositories from %v for %v: %v", repositories, distroStruct.Name(), err)
+		if err != nil {
+			// There is no point in running the tests without having repositories, but we can still run tests
+			// for the remaining distros
+			continue
+		}
+		// TODO: wait for @teg to implement GetArches() method and then loop over them, meanwhile at least hard code them
+		for _, archStr := range []string{"x86_64", "aarch64"} {
+			arch, err := distroStruct.GetArch(archStr)
+			assert.Nilf(t, err, "Failed to GetArch from %v structure: %v", distroStruct.Name(), err)
+			// TODO: loop over image types
+			imgType, err := arch.GetImageType("ami")
+			assert.Nilf(t, err, "Failed to GetImageType for %v on %v: %v", distroStruct.Name(), arch.Name(), err)
+
+			// TODO: check also base packages
```

Comment:
Same, drop TODO or add the check, I'd say (or is there a reason not to)?

Reply:
Nope, it wasn't my intention to submit it like this, but I have a bash script that downloads this branch and run the tests and I wanted to run it asap, so I just decided to push unfinished work. Sorry about that. It is finished now.

Reply:
Ah, got it. No problem.